### ISSUE TITLE
Blocks are not persisted on save action and clicking on add icon triggers a save warning

### ISSUE THREAD

**Issue by ichim-david** [collective/volto-dropdownmenu#10]
With a simple volto app created and using only volto-dropdown menu as seen in the following package.json
https://gist.github.com/ichim-david/d42165da776faa6f1774bbc57e9243b4

When I try to add a block as soon as I click on the add icon I get a save action as seen in the screenshot
![save-triggered-on-click](https://user-images.githubusercontent.com/152852/105884748-f4ef5180-6010-11eb-90ce-58d4ee091d16.png)

I then add a listing block and I hit the save icon trying to save
![save-from-save-icon](https://user-images.githubusercontent.com/152852/105885049-4ac3f980-6011-11eb-8558-94846311092c.png)

Then I add a new menu item
![new-menu-item-added](https://user-images.githubusercontent.com/152852/105885086-56afbb80-6011-11eb-978d-90e5e7664558.png)

And when I go to check the previous menu item the block data is gone both from edit on missing in view
![block-is-now-missing](https://user-images.githubusercontent.com/152852/105885140-65966e00-6011-11eb-8e38-9cb1a14ff432.png)

**Comment by ichim-david** [comment]
@nzambello I know that package.json uses volto 9 however I have another project where I use 10.9 and it behaves the same.
There I have a bunch of volto-slate plus volto-slate-addons and I though that maybe it interfered with this package given that
it's another text editor however even on this clean project it behaves the same.

**Comment by ichim-david** [comment]
@giuliaghisini ^

**Comment by giuliaghisini** [comment]
> @giuliaghisini ^

because you've to use this volto's branch **formFnOnUpdateFormData** that is waiting to be merged in master

**Comment by ichim-david** [comment]
@giuliaghisini I don't think that such a branch is made public, here is what I'm seeing on the branches page
<img width="948" alt="public-branches" src="https://user-images.githubusercontent.com/152852/106445887-415ee500-6488-11eb-9c6d-e154861e043c.PNG">

**Comment by ichim-david** [comment]
@giuliaghisini on further inspection of your comment I realized that you were referring to a branch within volto core
https://github.com/plone/volto/tree/formFnOnUpdateFormData
Still there is no mention of such a branch needed within ticket package.

I will try to have a look and see if I can figure out further what is happening.
